src/Features/Core/Portable/Completion/Providers/ImportCompletionProvider/ImportCompletionItem.cs

```diff
@@ -128,9 +130,21 @@ private static string GetFullyQualifiedName(string namespaceName, string typeNam
         private static ISymbol? GetSymbol(CompletionItem item, Compilation compilation)
         {
             // If we have SymbolKey data (i.e. this is an extension method item), use it to recover symbol
-            if (item.Properties.TryGetValue(SymbolKeyData, out var symbolId))
+            if (item.Properties.TryGetValue(MethodKey, out var methodSymbolKey))
             {
-                return SymbolKey.ResolveString(symbolId, compilation).GetAnySymbol();
+                var methodSymbol = SymbolKey.ResolveString(methodSymbolKey, compilation).GetAnySymbol() as IMethodSymbol;
+
+                // Get reduced extension method symbol for the given receiver type.
+                if (item.Properties.TryGetValue(ReceiverKey, out var receiverTypeKey))
+                {
+                    var receiverTypeSymbol = SymbolKey.ResolveString(receiverTypeKey, compilation).GetAnySymbol() as ITypeSymbol;
+                    if (receiverTypeSymbol != null)
+                    {
+                        return methodSymbol?.ReduceExtensionMethod(receiverTypeSymbol) ?? methodSymbol;
```

Comment:
in the case that methodSymbol is null, we end up doign nothing, so just front load that check up front.